ra2ce/graph/networks_utils.py

```diff
@@ -1131,8 +1132,154 @@ def simplify_graph_count(complex_graph: nx.Graph) -> nx.Graph:
     return simple_graph
 
 
+def simplify_graph(G, strict=True, remove_rings=True, track_merged=False):
+    """
+    this is an OSMNX function with the same name, modified to check if geometry arrtibute exists in the nodes of each
+    path to simplify.
+
+        Simplify a graph's topology by removing interstitial nodes.
+
+        Simplifies graph topology by removing all nodes that are not intersections
+        or dead-ends. Create an edge directly between the end points that
+        encapsulate them, but retain the geometry of the original edges, saved as
+        a new `geometry` attribute on the new edge. Note that only simplified
+        edges receive a `geometry` attribute. Some of the resulting consolidated
+        edges may comprise multiple OSM ways, and if so, their multiple attribute
+        values are stored as a list. Optionally, the simplified edges can receive
+        a `merged_edges` attribute that contains a list of all the (u, v) node
+        pairs that were merged together.
+
+        Parameters
+        ----------
+        G : networkx.MultiDiGraph
+            input graph
+        strict : bool
+            if False, allow nodes to be end points even if they fail all other
+            rules but have incident edges with different OSM IDs. Lets you keep
+            nodes at elbow two-way intersections, but sometimes individual blocks
+            have multiple OSM IDs within them too.
+        remove_rings : bool
+            if True, remove isolated self-contained rings that have no endpoints
+        track_merged : bool
+            if True, add `merged_edges` attribute on simplified edges, containing
+            a list of all the (u, v) node pairs that were merged together
+
+        Returns
+        -------
+        G : networkx.MultiDiGraph
+            topologically simplified graph, with a new `geometry` attribute on
+            each simplified edge
+    """
+    if "simplified" in G.graph and G.graph["simplified"]:  # pragma: no cover
+        msg = "This graph has already been simplified, cannot simplify it again."
+        raise GraphSimplificationError(msg)
+
+    utils.log("Begin topologically simplifying the graph...")
+
+    # define edge segment attributes to sum upon edge simplification
+    attrs_to_sum = {"length", "travel_time"}
+
+    # make a copy to not mutate original graph object caller passed in
+    G = G.copy()
```

Comment:
you are failing to do that the moment that you reassign it to t he same variable (please avoid one letter variables / arguments)
```suggestion
    # make a copy to not mutate original graph object caller passed in
    _result_graph = G.copy()
```